### PR TITLE
fix: pass env when getting param announcement (#11812)

### LINKED ISSUE #11812: 2.6 RC1 - sidecar plugin - spec.parameters.dynamic.command do not have in env ARGOCD_ENV_* variables
### sidecar plugin - spec.parameters.dynamic.command do not have in env ARGOCD_ENV_* variables
**_generate.command is working as expected_** 

**kind: ConfigManagementPlugin**
```
apiVersion: argoproj.io/v1alpha1
kind: ConfigManagementPlugin
metadata:
  name: miracle-plugin
  namespace: argocd
spec:
  generate:
    command: ["miracle-plugin.sh"]
  parameters:
    dynamic:
      command: ["miracle-plugin-param.sh"]
  discover:
    find:
      command: [sh, -c, find . -name *.yaml]
```
**miracle-plugin-param.sh**
```
set -eo pipefail

me=`basename "$0"`
helm=()
conf=()
kustomize=()
valuesfile=''
conffile=''

cat <<EOF >> /home/argocd/miracle-param.log
------ $( date +"%Y-%m-%dT%H:%M:%S%z" ) ------
ARGOCD_ENV_CLUSTER == $ARGOCD_ENV_CLUSTER
ARGOCD_ENV_ENV     == $ARGOCD_ENV_ENV
ARGOCD_ENV_APPNS   == $ARGOCD_ENV_APPNS
--------------------------------------
EOF

for file in base/chart/* targets/$ARGOCD_ENV_CLUSTER/$ARGOCD_ENV_ENV/$ARGOCD_ENV_APPNS/chart/* ; do
  if [[ $file =~ value ]]; then
    helm+=($file)
  fi
done

for file in base/c*.yaml targets/$ARGOCD_ENV_CLUSTER/$ARGOCD_ENV_ENV/$ARGOCD_ENV_APPNS/c*.yaml ; do
  if [[ $file =~ c*.yaml ]]; then
    conf+=($file)
  fi
done

for line in ${helm[@]}; do
  if [[ $line =~ value ]]; then
    valuesfile+=" $(pwd)/$line "
  fi
done

for line in ${conf[@]}; do
  if [[ $line =~ c*.yaml ]]; then
    conffile+=" $(pwd)/$line "
  fi
done

HELMPARAMS=$(yq e -o=p $conffile $valuesfile | jq --compact-output --slurp --raw-input '
  {
    name: "helm-parameters",
    title: "Helm Parameters",
    tooltip: "Parameters to override when generating manifests with Helm",
    collectionType: "map",
    map: (. | split("\n") | map(select(. != "") | capture("(?<key>.*) = (?<value>.*)")) | from_entries)
  }' )

echo $HELMPARAMS | jq --slurp
```
image

image

image

**miracle-plugin.sh**
**_generate.command is working as expected_** 

```
set -eo pipefail

me=`basename "$0"`
helm=()
conf=()
kustomize=()
valuesfile=''
conffile=''

for file in base/chart/* targets/$ARGOCD_ENV_CLUSTER/$ARGOCD_ENV_ENV/$ARGOCD_ENV_APPNS/chart/* ; do
  if [[ $file =~ value ]]; then
    helm+=($file)
  fi
done

for file in base/c*.yaml targets/$ARGOCD_ENV_CLUSTER/$ARGOCD_ENV_ENV/$ARGOCD_ENV_APPNS/c*.yaml ; do
  if [[ $file =~ c*.yaml ]]; then
    conf+=($file)
  fi
done

for line in ${helm[@]}; do
  if [[ $line =~ value ]]; then
    valuesfile+=" -f $(pwd)/$line "
  fi
done

for line in ${conf[@]}; do
  if [[ $line =~ c*.yaml ]]; then
    conffile+=" $(pwd)/$line "
  fi
done

for file in targets/$ARGOCD_ENV_CLUSTER/$ARGOCD_ENV_ENV/$ARGOCD_ENV_APPNS/kustomize/* ; do
  if [[ $file =~ kustomization ]]; then
    kustomize+=($file)
  fi
done

if [ ${#kustomize[@]} == 1 ]; then
  arg="${kustomize[0]%/*}/"
  kustomize build $(pwd)/$arg > /home/argocd/${ARGOCD_ENV_CLUSTER}_${ARGOCD_ENV_ENV}_${ARGOCD_ENV_APPNS}.yaml
  kustomize build $(pwd)/$arg
  kustcmd+="kustomize build $(pwd)/$arg"
fi

export YQ_ENV_HELM_REPO_URL=$(yq ea '. as $item ireduce ({}; . * $item )'  $conffile | yq '.helm.repo')
export YQ_ENV_HELM_RELEASE_NAME=$(yq ea '. as $item ireduce ({}; . * $item )'  $conffile | yq '.helm.release')
export YQ_ENV_HELM_CHART_VERSION=$(yq ea '. as $item ireduce ({}; . * $item )'  $conffile | yq '.helm.version')
export YQ_ENV_HELM_CHART_NAME=$(yq ea '. as $item ireduce ({}; . * $item )'  $conffile | yq '.helm.chart')

if [ ! -z "${valuesfile}" ]; then
            helm template --insecure-skip-tls-verify --version $YQ_ENV_HELM_CHART_VERSION --repo $YQ_ENV_HELM_REPO_URL --namespace $ARGOCD_APP_NAMESPACE $YQ_ENV_HELM_CHART_NAME --name-template=$YQ_ENV_HELM_RELEASE_NAME $valuesfile >> /home/argocd/${ARGOCD_ENV_CLUSTER}_${ARGOCD_ENV_ENV}_${ARGOCD_ENV_APPNS}.yaml
            helm template --insecure-skip-tls-verify --version $YQ_ENV_HELM_CHART_VERSION --repo $YQ_ENV_HELM_REPO_URL --namespace $ARGOCD_APP_NAMESPACE $YQ_ENV_HELM_CHART_NAME --name-template=$YQ_ENV_HELM_RELEASE_NAME $valuesfile
  helmcmd+="helm template --insecure-skip-tls-verify --version $YQ_ENV_HELM_CHART_VERSION --repo $YQ_ENV_HELM_REPO_URL --namespace $ARGOCD_APP_NAMESPACE $YQ_ENV_HELM_CHART_NAME --name-template=$YQ_ENV_HELM_RELEASE_NAME $valuesfile"
fi

cat <<EOF >> /home/argocd/miracle.log
------ $( date +"%Y-%m-%dT%H:%M:%S%z" ) ------
ARGOCD_ENV_APPNAME   == $ARGOCD_ENV_APPNAME
ARGOCD_ENV_CLUSTER   == $ARGOCD_ENV_CLUSTER
ARGOCD_APP_NAME      == $ARGOCD_APP_NAME
ARGOCD_APP_NAMESPACE == $ARGOCD_APP_NAMESPACE

     helm == ${helm[@]}
     conf == ${conf[@]}
kustomize == ${kustomize[@]}

command executed:
$helmcmd
$kustcmd
--------------------------------------
EOF
```

### PR DESCRIPTION
**Pull request by crenshaw-dev**
Fixes #11812

### CODE REVIEW

File: cmpserver/plugin/plugin.go [hunk 1]
```diff
@@ -358,7 +358,7 @@ func (s *Service) GetParametersAnnouncement(stream apiclient.ConfigManagementPlu
 		return fmt.Errorf("illegal appPath: out of workDir bound")
 	}
 
-	repoResponse, err := getParametersAnnouncement(bufferedCtx, appPath, s.initConstants.PluginConfig.Spec.Parameters.Static, s.initConstants.PluginConfig.Spec.Parameters.Dynamic)
+	repoResponse, err := getParametersAnnouncement(bufferedCtx, appPath, s.initConstants.PluginConfig.Spec.Parameters.Static, s.initConstants.PluginConfig.Spec.Parameters.Dynamic, metadata.Env)
```
Comment (leoluz) [inline cmpserver/plugin/plugin.go:361]: It seems that this is passing all repo-server environment variables to the cmp server. If this is really the case, it is maybe better to filter and just send the environments variables required by https://github.com/argoproj/argo-cd/issues/11812.

**Review by leoluz — CHANGES_REQUESTED** [review]

File: cmpserver/plugin/plugin.go (hunk 1 above)
Comment (crenshaw-dev) [reply cmpserver/plugin/plugin.go:361]: `metadata.Env` contains only the env vars send by the stream request initiator, in this case `populatePluginAppDetails` in reposerver/repository/repository.go. This does not include all repo-server environment variables.

**Review by leoluz — APPROVED** [review]
LGTM

**Pull request merged**